docs/deployment/resource-providers/mesos.md

```diff
@@ -26,226 +26,217 @@ under the License.
+variables and config files which are shipped to the worker nodes.
+
+See [Mesos Architecture](http://mesos.apache.org/documentation/latest/architecture/) for a more details 
+on how frameworks are handled by Mesos.
+
+### Deploying User Libraries
+
+User libraries can be passed to the Mesos workers by placing them in Flink's `lib/` folder. This way, 
+they will be picked by Mesos' Fetcher and copied over into the worker's sandbox folders. Alternatively, 
+Docker containerization can be used as described in [Installing Flink on the Workers](#installing-flink-on-the-workers).
+
+### Installing Flink on the Workers
+
+Flink on Mesos offers two ways to distribute the Flink and user binaries within the Mesos cluster:
+1. **Using Mesos' Artifact Server**: The Artifact Server provides the resources which are moved by 
+   [Mesos' Fetcher](http://mesos.apache.org/documentation/latest/fetcher/) into the Mesos worker's 
+   [sandbox folders](http://mesos.apache.org/documentation/latest/sandbox/). It can be explicitly 
+   specified by setting [mesos.resourcemanager.tasks.container.type]({% link deployment/config.md %}#mesos-resourcemanager-tasks-container-type) 
+   to `mesos`. This is the default option and is used in the example commands of this page.
+2. **Using Docker containerization**: This enables the user to provide user libraries and other 
+   customizations as part of a Docker image. Docker utilization can be enabled by setting 
+   [mesos.resourcemanager.tasks.container.type]({% link deployment/config.md %}#mesos-resourcemanager-tasks-container-type) 
+   to `docker` and by providing the image name through [mesos.resourcemanager.tasks.container.image.name]({% link deployment/config.md %}#mesos-resourcemanager-tasks-container-image-name).
+
+### High Availability on Mesos
+
+You will need to run a service like Marathon or Apache Aurora which takes care of restarting the 
+JobManager process in case of node or process failures. In addition, Zookeeper needs to be configured 
+as described in the [High Availability section of the Flink docs]({% link deployment/ha/index.md %}).
 
 #### Marathon
 
-Marathon needs to be set up to launch the `bin/mesos-appmaster.sh` script.
-In particular, it should also adjust any configuration parameters for the Flink cluster.
+Marathon needs to be set up to launch the `bin/mesos-appmaster.sh` script. In particular, it should 
+also adjust any configuration parameters for the Flink cluster.
 
 Here is an example configuration for Marathon:
+{% highlight javascript %}
+{
+    "id": "flink",
+    "cmd": "/opt/flink-1.11.2/bin/mesos-appmaster.sh -Dmesos.resourcemanager.framework.user=root -Dmesos.resourcemanager.tasks.taskmanager-cmd=/opt/flink-1.11.2/bin/mesos-taskmanager.sh -Dmesos.master=master:5050 -Djobmanager.memory.process.size=1472m -Dtaskmanager.memory.process.size=3500m -Dtaskmanager.numberOfTaskSlots=2 -Dparallelism.default=2",
```

Comment:
Do we have to specify all the memory configurations or can we use the default values?

Reply:
Where comes `master:5050` from?